platform/api/esx/api.go

```diff
@@ -0,0 +1,759 @@
+	}
+
+	ovfHandler := object.NewOvfManager(a.client.Client)
+	cisp := types.OvfCreateImportSpecParams{
+		EntityName:         name,
+		OvfManagerCommonParams: types.OvfManagerCommonParams{
+			Locale:           "US"},
+		PropertyMapping: []types.KeyValue{},
+		NetworkMapping:  networkMap(finder, envelope),
+	}
+
+	descriptor, err := archive.ReadOvf("*.ovf")
+	if err != nil {
+		return nil, nil, fmt.Errorf("reading ovf: %v", err)
+	}
+	cisr, err := ovfHandler.CreateImportSpec(a.ctx, string(descriptor), resourcePool, datastore, cisp)
+	if err != nil {
+		return nil, nil, err
+	}
+	if cisr.Error != nil {
+		return nil, nil, errors.New(cisr.Error[0].LocalizedMessage)
+	}
+	return archive, cisr, nil
+}
+
+func (a *API) getServerDefaults() (*find.Finder, *object.Datacenter, *object.ResourcePool, *object.Datastore, object.NetworkReference, error) {
+	finder := find.NewFinder(a.client.Client, true)
+	datacenter, err := finder.DefaultDatacenter(a.ctx)
+	if err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+	finder.SetDatacenter(datacenter)
+	resourcePool, err := finder.DefaultResourcePool(a.ctx)
+	if err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+	datastore, err := finder.DefaultDatastore(a.ctx)
+	if err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+
+	defaultNetwork, err := finder.DefaultNetwork(a.ctx)
+	if err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+
+	return finder, datacenter, resourcePool, datastore, defaultNetwork, nil
+}
+
+func (a *API) uploadToResourcePool(archive *Archive, resourcePool *object.ResourcePool, cisr *types.OvfCreateImportSpecResult, folder *object.Folder) (*types.ManagedObjectReference, error) {
+	lease, err := resourcePool.ImportVApp(a.ctx, cisr.ImportSpec, folder, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	info, err := lease.Wait(a.ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var items []ovfFileItem
+
+	for _, device := range info.DeviceUrl {
+		for _, item := range cisr.FileItem {
+			if device.ImportKey != item.DeviceId {
+				continue
+			}
+
+			u, err := a.client.Client.ParseURL(device.Url)
+			if err != nil {
+				return nil, err
+			}
+
+			i := ovfFileItem{
+				url:  u,
+				item: item,
+				ch:   make(chan progress.Report),
+			}
+
+			items = append(items, i)
+		}
+	}
+
+	upd := newLeaseUpdater(a.client.Client, lease, items)
+	defer upd.Done()
+
+	for _, i := range items {
+		err = a.upload(archive, lease, i)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	err = lease.HttpNfcLeaseComplete(a.ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &info.Entity, nil
+}
+
+func networkMap(finder *find.Finder, e *ovf.Envelope) (p []types.OvfNetworkMapping) {
+	ctx := context.TODO()
+
+	networks := map[string]string{}
+
+	if e.Network != nil {
+		for _, net := range e.Network.Networks {
+			networks[net.Name] = net.Name
+		}
+	}
+
+	for src, dst := range networks {
+		if net, err := finder.Network(ctx, dst); err == nil {
```

Comment:
It looks like this is the only place `ctx` is used, maybe just put `context.TODO()` inline here?